core/services/job/models.go

```diff
@@ -425,13 +425,13 @@ type KeeperSpec struct {
 
 type VRFSpec struct {
 	ID                       int32
-	CoordinatorAddress       ethkey.EIP55Address  `toml:"coordinatorAddress"`
-	PublicKey                secp256k1.PublicKey  `toml:"publicKey"`
-	MinIncomingConfirmations uint32               `toml:"minIncomingConfirmations"`
-	ConfirmationsEnv         bool                 `toml:"-"`
-	EVMChainID               *utils.Big           `toml:"evmChainID"`
-	FromAddress              *ethkey.EIP55Address `toml:"fromAddress"`
-	PollPeriod               time.Duration        `toml:"pollPeriod"` // For v2 jobs
+	CoordinatorAddress       ethkey.EIP55Address   `toml:"coordinatorAddress"`
+	PublicKey                secp256k1.PublicKey   `toml:"publicKey"`
+	MinIncomingConfirmations uint32                `toml:"minIncomingConfirmations"`
+	ConfirmationsEnv         bool                  `toml:"-"`
+	EVMChainID               *utils.Big            `toml:"evmChainID"`
+	FromAddresses            []ethkey.EIP55Address `toml:"fromAddresses"`
```

Comment:
Should we update `ValidatedVRFSpec` to check that there is at least one address in this slice? 

Before this change, we would protect against a possible nil `FromAddress` by calling `SendingKeys()` and just using the first one in the listener (I think this was potentially a hidden bug though, because there's no guarantee that the first key in the `SendingKeys()` return value has the correct gas price for that keyhash).

Reply:
If FromAddresses is empty / nil it will round robin through all addresses on the node. For V1 the current specs don't define a FromAddress I think so we can't break compatibility by requiring one here

Reply:
Oh you're right, if `FromAddresses` is empty then `GetRoundRobinAddress` will round robin on all sending keys for the node. For V1 this is fine but for V2 we might be selecting a key with a wrong max gas price, so we should ensure that we set `FromAddresses` in the job spec for V2.